### PR TITLE
Allow custom credentials file override

### PR DESCRIPTION
**Pull request by hiimdoublej-swag**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,3 +19,8 @@ RUN             apk add --virtual .build-deps \
                 apk add --virtual .run-deps \
                     ca-certificates && \
                 apk del .build-deps
+
+COPY            entrypoint.sh setup_creds.py /src/
+
+ENTRYPOINT      ["/src/entrypoint.sh"]
```
Comment (WeiHsin-swag) [inline Dockerfile:25]: It seems we need a permission for executing this file?
```
RUN             chmod +x /src/entrypoint.sh
```

**Review by WeiHsin-swag — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (hiimdoublej-swag) [reply Dockerfile:25]: Yes, it was missing from the PR

File: entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env sh
+set -e
+python3 /src/setup_creds.py && gsutil "$@"
```
Comment (hiimdoublej-swag) [inline entrypoint.sh:3]: Should just change this file to be `gsutil` so everything else stays the same.

File: Dockerfile (hunk 1 above)
Comment (hiimdoublej-swag) [reply Dockerfile]: Updated, the file should have executing permissions. But I'm not sure where does GitHub show it.

**Review by allanlei — CHANGES_REQUESTED** [review]
Take a look at something like s6 or even maybe dockerize?

Overriding the `gsutil` path like that isn't the way to go

**Comment by allanlei** [comment]
Using dockerize, then it would be something like

```
ENTRYPOINT ["dockerize", "-template", ".boto.tmpl:.boto", "gsutil"]
```

**Comment by hiimdoublej-swag** [comment]
Updated to use `dockerize` and a template for `.boto`

**Review by zhang699 — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -21,4 +19,10 @@ RUN             apk add --virtual .build-deps \
...
                 apk add --virtual .run-deps \
                     ca-certificates && \
-                apk del .build-deps
+                apk del .build-deps && \
+                wget https://github.com/jwilder/dockerize/releases/download/${DOCKERIZE_VERSION}/dockerize-alpine-linux-amd64-${DOCKERIZE_VERSION}.tar.gz && \
+                tar -C /usr/local/bin -xzvf dockerize-alpine-linux-amd64-${DOCKERIZE_VERSION}.tar.gz && \
+                rm dockerize-alpine-linux-amd64-${DOCKERIZE_VERSION}.tar.gz
+
+ENTRYPOINT      ["dockerize", "-template", "/root/.boto.tmpl:/root/.boto", "gsutil"]
+CMD             ["version", "-l"]
```
Comment (allanlei) [inline Dockerfile:28]: These should probably move back to the original position since it is unlikely to change

**Review by allanlei — APPROVED** [review]

**Pull request merged**